libraries/hardwareInterfaces.js

```diff
@@ -634,10 +634,33 @@ exports.getDebug = function () {
     return globalVariables.debug;
 };
 
+let setHardwareInterfaceSettingsImpl = null;
+/**
+ * Updates the settings.json for a particular hardware interface, based on
+ * changes from the webFrontend.  Uses setHardwareInterfaceSettingsImpl to cut
+ * through a bunch of abstraction into the heart of the server
+ *
+ * @param {string} interfaceName - the folder name of the hardwareInterface
+ * @param {JSON} settings - JSON structure of the new settings to be written to settings.json
+ * @param {Array.<string>} limitToKeys - if provided, only affects the properties of settings whose keys are included in this array
+ * @param {successCallback} callback
+ */
+exports.setHardwareInterfaceSettings = function(interfaceName, settings, limitToKeys, callback) {
+    return setHardwareInterfaceSettingsImpl(interfaceName, settings, limitToKeys, callback);
+};
+
 /*
  ********** END API FUNCTIONS *********
  */
 
+/**
+ * Complement to setup() which is necessary due to the unique positioning of
+ * the setHardwareInterfaceSettings function
+ */
+exports.setHardwareInterfaceSettingsImpl = function(setHardwareInterfaceSettings) {
+    setHardwareInterfaceSettingsImpl = setHardwareInterfaceSettings;
+};
```

Comment:
Interesting structure but it makes sense why this is needed

Reply:
Yeah it's adding a couple of strands of spaghetti